### PR TITLE
Add e2e test for cluster creation with AWS KMS

### PR DESCRIPTION
**Pull request by muraee**
**What this PR does / why we need it**:
Add e2e test for cluster creation with AWS KMS

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes #[HOSTEDCP-386](https://issues.redhat.com/browse/HOSTEDCP-386)

**Checklist**
- [ ] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

File: test/e2e/create_cluster_test.go [hunk 1]
```diff
@@ -73,6 +73,48 @@ func TestCreateCluster(t *testing.T) {
 	e2eutil.EnsureNodeCommunication(t, ctx, client, hostedCluster)
 }
```
Comment (enxebre) [inline test/e2e/create_cluster_test.go:74]: please separate vendor go.* changes into its own commit so this easier for reviewers

File: control-plane-operator/controllers/hostedcontrolplane/kas/aws_kms.go [hunk 2]
```diff
@@ -216,6 +217,10 @@ func buildKASContainerAWSKMSTokenMinter(image string) func(*corev1.Container) {
 			fmt.Sprintf("--token-file=%s", path.Join(awsKMSVolumeMounts.Path(c.Name, kasVolumeAWSKMSCloudProviderToken().Name), "token")),
 			fmt.Sprintf("--kubeconfig=%s", path.Join(awsKMSVolumeMounts.Path(c.Name, kasVolumeLocalhostKubeconfig().Name), KubeconfigKey)),
 		}
+		c.Resources.Requests = corev1.ResourceList{
+			corev1.ResourceCPU:    resource.MustParse("10m"),
```
Comment (enxebre) [inline control-plane-operator/controllers/hostedcontrolplane/kas/aws_kms.go:221]: where are these values coming from?

File: test/e2e/create_cluster_test.go [hunk 3]
```diff
@@ -73,6 +73,48 @@ func TestCreateCluster(t *testing.T) {
...
+	numNodes := clusterOpts.NodePoolReplicas * int32(len(clusterOpts.AWSPlatform.Zones))
+	e2eutil.WaitForNReadyNodes(t, testContext, guestClient, numNodes, hostedCluster.Spec.Platform.Type)
+
+	// Wait for the rollout to be complete
+	t.Logf("Waiting for cluster rollout. Image: %s", globalOpts.LatestReleaseImage)
+	e2eutil.WaitForImageRollout(t, testContext, client, hostedCluster, globalOpts.LatestReleaseImage)
+
+	err = client.Get(testContext, crclient.ObjectKeyFromObject(hostedCluster), hostedCluster)
+	g.Expect(err).NotTo(HaveOccurred(), "failed to get hostedcluster")
+
```
Comment (enxebre) [inline test/e2e/create_cluster_test.go:113]: this test is actually not validating the data is being encrypted e.g. https://kubernetes.io/docs/tasks/administer-cluster/kms-provider/#verifying-that-the-data-is-encrypted

As it's right now, it only validate the input does not break the workflow.
We can do that in a follow up PR though.

**Comment by enxebre** [comment]
Thanks @muraee! this is awesome. Dropped some comments.
PTAL @sjenning @csrwng

File: control-plane-operator/controllers/hostedcontrolplane/kas/aws_kms.go (hunk 2 above)
Comment (muraee) [reply control-plane-operator/controllers/hostedcontrolplane/kas/aws_kms.go:221]: These are the same values used for the `token-minter`  container here https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L2201-L2205

**Pull request merged**